### DISCUSSION TITLE
Cannot clone private repo from Codeberg

### DISCUSSION THREAD

**Discussion by pfr-dev** [zeapo/Android-Password-Store#1346] [Q&A] [answered]
I have my `.password-store` in a private codeberg repo. When I try to clone the repo using the app I get:
```
An error occurred during a Git operation
https://codeberg.org/my/private/.password-store: Authentication is required but no CredentialsProvider has been registered
```
Now, obviously I have not been able to enter my passphrase for my private repo anywhere and this is clearly the problem. 

Where in the app can I register my codeberg password?? I have entered my Username and Email address, but there is no input box for password that I can find. 

Cheers

**Comment by msfjarvis** [comment, marked as answer]
Did you select the correct authentication method in the git server settings screen? If you select password there then you will be prompted for a password during clone. The username and email that you provided are for your local Git author configuration, so for cloning your repository you will have to provide your username within the URL itself, like so : `https://<username>@codeberg.org/my/private/.password-store` where `<username>` will be replaced with your Codeberg username.

**Reply by pfr-dev** [reply]
> you will have to provide your username within the URL itself, like so : `https://<username>@codeberg.org/my/private/.password-store`

Ahhhh damn ok, this worked. Thank you!
